scripts/cargo-test.sh

```diff
@@ -1,4 +1,14 @@
-#!/usr/bin/env sh
+#!/usr/bin/env bash
+
+cleanup_handler() {
```

Comment:
The composer now also has an option .with_prune() that when enabled, will clean up any existing containers before starting a new test. The upside of that is that if there is an issue, we can docker logs it. With this change, we kill the container (and thus lose the logs). Would that help your situation?

Reply:
I was thinking about the logs, but as they are captured by the test, I hoped that we can forget about them.

> The composer now also has an option .with_prune() that when enabled, will clean up any existing containers before starting a new test.

I dunno, it seems that it is only a marginal improvement as a consecutive run will prune containers anyway (and we will lose logs). I would say that it will be easier from the developers' perspective as she can just re-run in Jenkins. I think that really robust one will be the 3rd option from the description:

> prefix all names in tests with a selector passed as an environment variable to allow non conflicting names for parallel test runs + being able to select out containers related to this run in the handler.

Which I can actually try to implement, if you see any value in it.

Reply:
hmm might not necessarily loose the logs in either case, we can access them via journalctl -u docker?

Reply:
> hmm might not necessarily loose the logs in either case, we can access them via journalctl -u docker?

Yes, with little bit of configuration, thanks. It gives you only `dockerd` logs by default. But one can configure [journald logging driver](https://docs.docker.com/config/containers/logging/journald/) quite easily if we feel the need for it.